src/Connections.Abstractions/IConnection.cs

```diff
@@ -0,0 +1,17 @@
+using System.Collections.Generic;
+using System.IO.Pipelines;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http.Features;
+
+namespace Microsoft.AspNetCore.Connections
+{
+    public interface IConnection
+    {
+        IDuplexPipe Transport { get; }
+        IFeatureCollection Features { get; }
+
+        Task StartAsync();
+        Task StartAsync(TransferFormat transferFormat);
```

Comment:
TransferFormat.Text in general purpose connection interface? :/

Reply:
This is for preview2. See the comment above there are plans to not have this interface for rc1

Reply:
And `StartAsync` is going to be handled on the client by a separate "Connection Factory" component which won't need to be in these abstractions, so the Http.Connections.Client can be the only one that has to care about TransferFormat (aside from the feature and enum _existing_ in the abstractions)